File: python/7 kyu/41.py
from math import gcd

# https://www.codewars.com/kata/576400f2f716ca816d001614/train/python

# Passed

def reduce_fraction(fraction):
    greatest_common_divisor = gcd(*fraction)
    if greatest_common_divisor != 1:
        numerator = int(fraction[0]/greatest_common_divisor)
        denominator = int(fraction[1]/greatest_common_divisor)
        return reduce_fraction([numerator, denominator])

    return fraction

output = reduce_fraction([9, 27])
print(output)

File: nodejs/7 kyu/41.js
// https://www.codewars.com/kata/569b5cec755dd3534d00000f/train/javascript

// Passed

function newAvg(arr, newavg) {
  const totalDonationsTillNow = arr.reduce((prev, curr) => prev + curr, 0);
  const possibleTotalDonations = arr.length + 1;
  const donationToBeMade =
    newavg * possibleTotalDonations - totalDonationsTillNow;

  if (donationToBeMade > 0)
    return donationToBeMade % 1 !== 0
      ? Math.ceil(donationToBeMade)
      : donationToBeMade;

  throw new Error("Expected New Average is too low");
}

const output = newAvg([14, 30, 5, 7, 9, 11, 16], 90, 628);
console.log(output);
